torch/utils/dlpack.py

```diff
@@ -25,3 +15,38 @@
 The dlpack shares the tensors memory.
 Note that each dlpack can only be consumed once.
 """)
+
+
+def from_dlpack(ext_tensor: Any) -> torch.Tensor:
+    """from_dlpack(ext_tensor) -> Tensor
+
+    Convers a tensor from a external library into a ``torch.Tensor``
+    by means of the ``__dlpack__`` protocol.
+
+    The tensor will share the memory with the object represented
+    in the dlpack.
+
+    In order to keep backward compatibility, this function also admits
+    to pass a dlpack capsule object.
+    Note that each dlpack capsule can only be consumed once.
+
+    Args:
+        ext_tensor (object with __dlpack__ attribute or dlpack capsule):
+            The tensor from an external library that will be converted
+            to a PyTorch one.
+    """
+    if hasattr(ext_tensor, '__dlpack__'):
+        device = ext_tensor.__dlpack_device__()
+        # device is either CUDA or ROCm, we need to pass the current
+        # stream
+        if device[0] in (2, 10):
```

Comment:
What is the test for the device being 2 or 10? Are those just magic numbers defined by the DLPack specification?

Reply:
yes, I should change this to actual nemonics, sorry